frontend/src/main/scala/bloop/engine/tasks/Tasks.scala

```diff
@@ -134,3 +141,102 @@ object CompileTasks {
     state
   }
 }
+
+object RunTasks {
+  def run(state: State, project: Project, className: String, args: Array[String]): State = {
+    val classpath = project.classpath
+    val processConfig = ProcessConfig(project.javaEnv, classpath)
+    val exitCode = processConfig.runMain(className, args, state.logger)
+    val exitStatus = {
+      if (exitCode == ProcessConfig.EXIT_OK) ExitStatus.Ok
+      else ExitStatus.UnexpectedError
+    }
+
+    state.mergeStatus(exitStatus)
+  }
+
+  def findMainClasses(state: State, project: Project): Array[String] = {
+    import state.logger
+    import bloop.util.JavaCompat.EnrichOptional
+    val analysis = state.results.getResult(project).analysis().toOption match {
+      case Some(analysis: Analysis) => analysis
+      case _ =>
+        logger.warn(s"`Run` is triggered but no compilation detected from '${project.name}'.")
+        sbt.internal.inc.Analysis.empty
+    }
+
+    val mainClasses = analysis.infos.allInfos.values.flatMap(_.getMainClasses)
+    logger.debug(s"Found ${mainClasses.size} main classes${mainClasses.mkString(": ", ", ", ".")}")
+    mainClasses.toArray
+  }
+}
+
+object TestTasks {
```

Comment:
I didn't really mean this when I said putting everything in the `Tasks` class. What I meant is to only have one object `Tasks` that has all these methods. Right now, it doesn't make much sense that we have to remember the prefix of the object where we call `compile`, `test` and `run` to be `CompileTasks`, `TestTasks` and `RunTasks`. I'd rather have `Tasks.compile`, `Tasks.test` and `Tasks.run`. I think it makes for a better API.